Applications/Spire/Source/Ui/Ui.cpp

```diff
@@ -18,6 +18,22 @@ void Spire::draw_border(const QRect& region, const QColor& color,
   painter->restore();
 }
 
+QPropertyAnimation* Spire::fade_window(QObject* target, bool reverse,
+    int fade_speed_ms) {
```

Comment:
When you have the choice to define the API, use a ```boost::posix_time::time_duration``` for time durations. Then convert it to an int when you need to pass it to Qt.